apps/synapse-oauth-signin/src/mocks/handlers.ts

```diff
@@ -1,6 +1,14 @@
 import { rest } from 'msw'
 import mockOauthClient from './MockOAuthClient'
 
+let hasConsented = false
+
+export function resetConsentedInMockService(newValue: boolean) {
+  hasConsented = newValue
+}
```

Comment:
Store the status of whether the user has consented, and allow resetting it, so that the mock server behavior changes appropriately as tests interact with it.